pyscriptjs/src/plugins/pyterminal.ts

```diff
@@ -0,0 +1,123 @@
+import type { PyScriptApp } from '../main';
+import type { AppConfig } from '../pyconfig';
+import { Plugin } from '../plugin';
+import { getLogger } from '../logger';
+import { type Stdio } from '../stdio';
+
+const logger = getLogger('py-terminal');
+
+export class PyTerminalPlugin extends Plugin {
+    app: PyScriptApp;
+
+    constructor(app: PyScriptApp) {
+        super();
+        this.app = app;
+    }
+
+    configure(config: AppConfig) {
+        // validate the terminal config and handle default values
+        const t = config.terminal;
```

Comment:
Just a nit, but would be nice to have a more discriptive name although the code is small enough to not be too confusing 😄

Reply:
I'm following [this rule](https://twitter.com/unclebobmartin/status/1023187234915266561) here. The point is precisely that the code is short, and having a longer name would make it harder to read IMHO.
The reason why I introduce the extra var was to avoid this:
```typescript
        // validate the terminal config and handle default values
        if (config.terminal !== undefined &&
            config.terminal !== true &&
            config.terminal !== false &&
            config.terminal !== "auto") {
            // XXX this should be a UserError as soon as we have it
            const got = JSON.stringify(config.terminal);
            throw new Error('Invalid value for config.terminal: the only accepted'  +
                            `values are true, false and "auto", got "${got}".`);
        }
        if (config.terminal === undefined) {
            config.terminal = "auto"; // default value
        }
```

That said, I'm happy to change `t` into a longer name if you think it leads to more readable code: what do you think it should be?
I'm serious here: an external reviewer is a better judge w.r.t readability than the original author of the code, so happy to hear your suggestions.

Reply:
in that part there is no `t` anymore